crates/paimon/src/spec/schema_change.rs

```diff
@@ -264,185 +271,181 @@ mod tests {
-        assert_eq!(
-            schema_change,
-            SchemaChange::update_column_comment("col4".to_string(), "col4 field".to_string())
-        );
-        // SchemaChange: UpdateColumnsComment
-        let schema_change = SchemaChange::update_columns_comment(
-            vec!["col5".to_string(), "f1".to_string()],
-            "col5 f1 field".to_string(),
-        );
-        let json = serde_json::to_string(&schema_change).unwrap();
-        let schema_change = serde_json::from_str::<SchemaChange>(&json).unwrap();
-        assert_eq!(
-            schema_change,
-            SchemaChange::update_columns_comment(
-                vec!["col5".to_string(), "f1".to_string()],
-                "col5 f1 field".to_string()
-            )
-        );
+        let json_data = r#"
+        [
+          {
+            "setOption": {
+              "key": "snapshot.time-retained",
+              "value": "2h"
+            }
+          },
+          {
+            "removeOption": {
+              "key": "compaction.max.file-num"
+            }
+          },
+          {
+            "updateComment": {
+              "comment": "table.comment"
+            }
+          },
+          {
+            "addColumn": {
+              "fieldName": "col1",
+              "dataType": {
+                "Int": {
+                  "nullable": true
+                }
+              },
+              "description": "col1_description",
+              "move": {
+                "fieldName": "col1_first",
+                "referencedFieldName": null,
+                "type": "FIRST"
+              }
+            }
+          },
+          {
+            "renameColumn": {
+              "fieldName": "col3",
+              "newName": "col3_new_name"
+            }
+          },
+          {
+            "dropColumn": {
+              "fieldName": "col1"
+            }
+          },
+          {
+            "updateColumnType": {
+              "fieldName": "col14",
+              "dataType": {
+                "Double": {
+                  "nullable": true
+                }
+              }
+            }
+          },
+          {
+            "updateColumnPosition": {
+              "move": {
+                "fieldName": "col4_first",
+                "referencedFieldName": null,
+                "type": "FIRST"
+              }
+            }
+          },
+          {
+            "updateColumnNullability": {
+              "fieldName": [
+                "col5",
+                "f2"
+              ],
+              "nullable": false
+            }
+          },
+          {
+            "updateColumnComment": {
+              "fieldNames": [
+                "col5",
+                "f1"
+              ],
+              "newDescription": "col5 f1 field"
+            }
+          }
+        ]"#;
+
+        let schema_changes: Vec<SchemaChange> = serde_json::from_str(json_data).unwrap();
+
+        for change in schema_changes {
```

Comment:
Hi, `SchemaChange` has implemented `Eq`, so we don't need to check it this way. Instead, we can:

```rust
assert_eq!(
    schema_changes, 
    vec![
        SchemaChange::SetOption {
            key: "snapshot.time-retained".to_string, 
            value: "2h".to_string()
        },
        ...
    ]);
```